plugins/woocommerce/src/Admin/Notes/NoteTraits.php

```diff
@@ -191,4 +199,65 @@ public static function has_note_been_actioned() {
 
 		return false;
 	}
+
+/**
+	 * Update a note field of note1 if it's different from note2 with getter and setter.
+	 *
+	 * @param Note   $note1 Note to update.
+	 * @param Note   $note2 Note to compare against.
+	 * @param string $field_name Field to update.
+	 * @return bool True if the field was updated.
+	 */
+	private static function update_note_field_if_changed( $note1, $note2, $field_name ) {
+		// We need to serialize the stdObject to compare it.
+		$note1_field_value = self::possibly_convert_object_to_array(
+			call_user_func( array( $note1, 'get_' . $field_name ) )
+		);
+		$note2_field_value = self::possibly_convert_object_to_array(
+			call_user_func( array( $note2, 'get_' . $field_name ) )
+		);
+
+		if ( 'actions' === $field_name ) {
+			// We need to individually compare the action fields because action object from db is different from action object of note.
+			// For example, action object from db has "id".
+			$diff        = array_udiff(
+				$note1_field_value,
+				$note2_field_value,
+				function( $action1, $action2 ) {
+					if ( $action1->name === $action2->name &&
```

Comment:
Would it be easier to just delete the fields that are irrelevant, prior to comparison?

Reply:
It's actually might need more steps because we need to delete properties and convert them to arrays before comparison, though it's very close. But I think the current approach is more explicit.